src/Components/Nav/Nav.js

```diff
@@ -0,0 +1,117 @@
+import { Link, useLocation } from 'react-router-dom';
+import { withRouter } from 'react-router-dom/cjs/react-router-dom.min';
+import styled from 'styled-components';
+
+function Nav() {
+  let locatioin = useLocation();
+
+  return (
+    <NavWrap>
+      <Logo location={locatioin} to="/">
+        가나다방
+      </Logo>
+      <Menu>
+        <NavUl>
+          <NavList>
+            <MenuLink location={locatioin} to="/map">
+              지도
+            </MenuLink>
+          </NavList>
+          <NavList>
+            <MenuLink location={locatioin} to="/">
+              분양
+            </MenuLink>
+          </NavList>
+          <NavList>
+            <MenuLink location={locatioin} to="/">
+              관심목록
+            </MenuLink>
+          </NavList>
+          <NavList>
+            <MenuLink location={locatioin} to="/">
+              방 내놓기
+            </MenuLink>
+          </NavList>
+          <NavList>
+            <MenuLink location={locatioin} to="/">
+              알림
+            </MenuLink>
+          </NavList>
+        </NavUl>
+        <LoginSignButton location={locatioin}>
+          로그인
+          <Bar location={locatioin}></Bar>
+          회원가입
+        </LoginSignButton>
+      </Menu>
+    </NavWrap>
+  );
+}
+
+const NavWrap = styled.div`
+  ${props => props.theme.flexSet('space-between')};
+  position: absolute;
+  width: 100%;
+  height: 8.5rem;
+  padding: 0 2rem 0 3rem;
+`;
+
+const Logo = styled(Link)`
+  font-size: 2.8rem;
+  font-family: 'Gugi', cursive;
+  color: ${props =>
+    props.location.pathname === '/' ? 'white' : props.theme.blue};
+`;
+
+const Menu = styled.div`
+  ${props => props.theme.flexSet()};
+`;
+
+const NavUl = styled.ul`
+  ${props => props.theme.flexSet()};
+  font-size: 1.5rem;
+`;
+
+const NavList = styled.li`
+  padding: 0 2rem;
+`;
+
+const MenuLink = styled(Link)`
+  color: ${props =>
+    props.location.pathname === '/' ? props.theme.white : props.theme.black};
```

Comment:
```suggestion
  color: ${({ location, theme }) => location.pathname === '/' ? theme.white : theme.black};
```
위에 말씀드린 내용 반영하면서 바뀔 부분이긴 하지만 함수 파라미터 자리에서 직접 구조분해 할당 가능합니다

Reply:
수정했습니다.